### PR TITLE
Changes for virus assembly pipeline

### PR DESCRIPTION
**Pull request by rhpvorderman**

### CODE REVIEW

File: spades.wdl [hunk 1]
```diff
@@ -0,0 +1,79 @@
+task spades {
+    String outputDir
+    String? preCommand
+    File? read1
+    File? read2
+    File? singleRead
```
Comment (DavyCats) [inline spades.wdl]: Shouldn't one of the inputs be mandatory?

File: spades.wdl [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    Boolean? plasmid
+    Boolean? ionTorrent
+    Boolean? onlyErrorCorrection
+    Boolean? onlyAssembler
+    Boolean? careful
+    Boolean? disableGzipOutput
+    Boolean? disableRepeatResolution
+    File? dataset
+    Int threads
+    Int memoryGb
```
Comment (DavyCats) [inline spades.wdl]: Should threads and memory really be mandatory? Seems more like a thing to make optional with a default.

File: spades.wdl [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        ${"-o " + outputDir} \
+        ${true="--sc" false="" singleCell} \
+        ${true="--meta" false="" metagenomic} \
+        ${true="--rna" false="" rna} \
+        ${true="--plasmid" false="" plasmid} \
+        ${true="--iontorrent" false="" ionTorrent} \
+        ${"--12 " + interlacedReads } \
+        ${"-1 " + read1 } \
+        ${"-2 " + read2 } \
+        ${"-s " + singleRead } \
```
Comment (DavyCats) [inline spades.wdl]: If this is replaced with `${true="-1" false="-s" defined(read2)} read1`, read1 can be made mandatory. If I'm not mistaken.

File: spades.wdl [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        ${"--sanger " + sangerReads } \
+        ${"--pacbio " + pacbioReads } \
+        ${"--nanopore " + nanoporeReads } \
+        ${"--tslr " + tslrContigs } \
+        ${"--trusted-contigs " + trustedContigs } \
+        ${"--untrusted-contigs" + untrustedContigs } \
```
Comment (DavyCats) [inline spades.wdl]: Missing space?

File: unicycler.wdl [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    File? short2
+    File? unpaired
+    File? long
+    String out
+    Int? verbosity
+    Int? minFastaLength
+    Int? keep
+    Boolean? vcf
+    Int threads
+    Int memory
```
Comment (DavyCats) [inline unicycler.wdl]: See comments regarding threads/memory above.

File: spades.wdl (hunk 1 above)
Comment (rhpvorderman) [reply spades.wdl]: Unfortunately, this is not possible. Unless you want to do some crazy logic magic in bash. (And I do not want to do that)

File: spades.wdl (hunk 2 above)
Comment (rhpvorderman) [reply spades.wdl]: There were some issues with running spades on the cluster and defining defaults

File: spades.wdl (hunk 3 above)
Comment (rhpvorderman) [reply spades.wdl]: :+1:

**Pull request closed without merging**